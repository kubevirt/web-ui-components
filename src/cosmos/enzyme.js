import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
