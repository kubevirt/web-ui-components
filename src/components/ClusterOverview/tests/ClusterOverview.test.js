import React from 'react';
import { shallow } from 'enzyme';

import { ClusterOverview } from '..';

describe('<ClusterOverview />', () => {
  it('renders correctly', () => {
    const component = shallow(<ClusterOverview />);
    expect(component).toMatchSnapshot();
  });
});
