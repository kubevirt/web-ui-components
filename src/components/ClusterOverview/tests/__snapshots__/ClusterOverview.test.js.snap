// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClusterOverview /> renders correctly with Provider 1`] = `
<div>
  <div>
    <h1>
      Cluster Dashboard
    </h1>
  </div>
  <div
    class="kubevirt-dashboard__body"
  >
    <div
      class="pf-l-grid"
    >
      <div
        class="pf-l-grid__item pf-m-12-col-on-sm pf-m-12-col-on-md pf-m-3-col-on-lg"
      >
        <div
          class="pf-l-grid"
        >
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-6-col-on-md pf-m-12-col-on-lg"
          >
            <div
              class="card-pf kubevirt-dashboard__card"
            >
              <div
                class="card-pf-heading kubevirt-dashboard__card-heading"
              >
                <h2
                  class="card-pf-title"
                >
                  Details
                </h2>
              </div>
              <div
                class="card-pf-body"
              >
                <dl>
                  <dt
                    class="kubevirt-detail__item-title"
                  >
                    Name
                  </dt>
                  <dd
                    class="kubevirt-detail__item-value"
                  >
                    <span>
                      clusterName.clusterDoman
                    </span>
                  </dd>
                  <dt
                    class="kubevirt-detail__item-title"
                  >
                    Provider
                  </dt>
                  <dd
                    class="kubevirt-detail__item-value"
                  >
                    <span>
                      AWS
                    </span>
                  </dd>
                  <dt
                    class="kubevirt-detail__item-title"
                  >
                    RHHI version
                  </dt>
                  <dd
                    class="kubevirt-detail__item-value"
                  >
                    <span>
                      1.0
                    </span>
                  </dd>
                  <dt
                    class="kubevirt-detail__item-title"
                  >
                    Openshift version
                  </dt>
                  <dd
                    class="kubevirt-detail__item-value"
                  >
                    <span>
                      v4.0.0
                    </span>
                  </dd>
                </dl>
              </div>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-6-col-on-md pf-m-12-col-on-lg"
          >
            <div
              class="card-pf kubevirt-dashboard__card"
            >
              <div
                class="card-pf-heading kubevirt-dashboard__card-heading"
              >
                <h2
                  class="card-pf-title"
                >
                  Cluster inventory
                </h2>
                <button
                  class="btn btn-link"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    class="fa fa-info-circle fa-2x"
                  />
                </button>
              </div>
              <div
                class="card-pf-body"
              >
                <div
                  class="kubevirt-inventory__row"
                  id="inventory-nodes"
                >
                  <div
                    class="kubevirt-inventory__row-title"
                  >
                    1 Nodes
                  </div>
                  <div
                    class="kubevirt-inventory__row-status"
                  >
                    <div
                      class="kubevirt-inventory__row-status-item"
                    >
                      <span
                        aria-hidden="true"
                        class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
                      />
                      <span
                        class="kubevirt-inventory__row-status-item-text"
                      >
                        1
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="kubevirt-inventory__row"
                  id="inventory-hosts"
                >
                  <div
                    class="kubevirt-inventory__row-title"
                  >
                    Hosts
                  </div>
                  <div>
                    <div
                      class="spinner spinner-md blank-slate-pf-icon"
                    />
                  </div>
                </div>
                <div
                  class="kubevirt-inventory__row"
                  id="inventory-pvcs"
                >
                  <div
                    class="kubevirt-inventory__row-title"
                  >
                    5 PVCs
                  </div>
                  <div
                    class="kubevirt-inventory__row-status"
                  >
                    <div
                      class="kubevirt-inventory__row-status-item"
                    >
                      <span
                        aria-hidden="true"
                        class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
                      />
                      <span
                        class="kubevirt-inventory__row-status-item-text"
                      >
                        5
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="kubevirt-inventory__row"
                  id="inventory-pods"
                >
                  <div
                    class="kubevirt-inventory__row-title"
                  >
                    1 Pods
                  </div>
                  <div
                    class="kubevirt-inventory__row-status"
                  >
                    <div
                      class="kubevirt-inventory__row-status-item"
                    >
                      <span
                        aria-hidden="true"
                        class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
                      />
                      <span
                        class="kubevirt-inventory__row-status-item-text"
                      >
                        1
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="kubevirt-inventory__row"
                  id="inventory-vms"
                >
                  <div
                    class="kubevirt-inventory__row-title"
                  >
                    2 VMs
                  </div>
                  <div
                    class="kubevirt-inventory__row-status"
                  >
                    <div
                      class="kubevirt-inventory__row-status-item"
                    >
                      <span
                        aria-hidden="true"
                        class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
                      />
                      <span
                        class="kubevirt-inventory__row-status-item-text"
                      >
                        2
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pf-l-grid__item pf-m-12-col-on-sm pf-m-12-col-on-md pf-m-3-col-on-lg"
      >
        <div
          class="pf-l-grid"
        >
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-6-col-on-md pf-m-12-col-on-lg"
          >
            <div
              class="card-pf kubevirt-dashboard__card"
            >
              <div
                class="card-pf-heading kubevirt-dashboard__card-heading"
              >
                <h2
                  class="card-pf-title"
                >
                  Cluster Events
                </h2>
                <button
                  class="btn btn-link"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    class="fa fa-info-circle fa-2x"
                  />
                </button>
              </div>
              <div
                class="card-pf-body kubevirt-events__body"
                id="events-body"
              >
                <div>
                  Requires component from tectonic
                </div>
              </div>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-6-col-on-md pf-m-12-col-on-lg"
          >
            <div
              class="card-pf kubevirt-dashboard__card"
            >
              <div
                class="card-pf-heading kubevirt-dashboard__card-heading"
              >
                <h2
                  class="card-pf-title"
                >
                  Top Consumers
                </h2>
                <button
                  class="btn btn-link"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    class="fa fa-info-circle fa-2x"
                  />
                </button>
              </div>
              <div
                class="card-pf-body"
              >
                <div>
                  <div
                    class="kubevirt-consumers__filters"
                  >
                    <div
                      class="btn-group btn-group-justified"
                    >
                      <div
                        class="dropdown btn-group btn-group-default"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="true"
                          class="kubevirt-dropdown dropdown-toggle btn btn-default"
                          id="filter-dropdown"
                          role="button"
                          type="button"
                        >
                          All 
                          <span
                            class="caret"
                          />
                        </button>
                        <ul
                          aria-labelledby="filter-dropdown"
                          class="dropdown-menu"
                          role="menu"
                        >
                          <li
                            class=""
                            role="presentation"
                          >
                            <a
                              href="#"
                              role="menuitem"
                              tabindex="-1"
                            >
                              All
                            </a>
                          </li>
                          <li
                            class=""
                            role="presentation"
                          >
                            <a
                              href="#"
                              role="menuitem"
                              tabindex="-1"
                            >
                              Pod
                            </a>
                          </li>
                          <li
                            class=""
                            role="presentation"
                          >
                            <a
                              href="#"
                              role="menuitem"
                              tabindex="-1"
                            >
                              VirtualMachine
                            </a>
                          </li>
                        </ul>
                      </div>
                    </div>
                    <div
                      class="btn-group btn-group-justified"
                    >
                      <div
                        class="dropdown btn-group btn-group-default"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="true"
                          class="kubevirt-dropdown dropdown-toggle btn btn-default"
                          id="sort-by-dropdown"
                          role="button"
                          type="button"
                        >
                          CPU 
                          <span
                            class="caret"
                          />
                        </button>
                        <ul
                          aria-labelledby="sort-by-dropdown"
                          class="dropdown-menu"
                          role="menu"
                        >
                          <li
                            class=""
                            role="presentation"
                          >
                            <a
                              href="#"
                              role="menuitem"
                              tabindex="-1"
                            >
                              CPU
                            </a>
                          </li>
                          <li
                            class=""
                            role="presentation"
                          >
                            <a
                              href="#"
                              role="menuitem"
                              tabindex="-1"
                            >
                              Memory
                            </a>
                          </li>
                          <li
                            class=""
                            role="presentation"
                          >
                            <a
                              href="#"
                              role="menuitem"
                              tabindex="-1"
                            >
                              Network
                            </a>
                          </li>
                          <li
                            class=""
                            role="presentation"
                          >
                            <a
                              href="#"
                              role="menuitem"
                              tabindex="-1"
                            >
                              Storage
                            </a>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                  <div
                    class="progress-container progress-description-left kubevirt-consumers__bar utilization-bar-pf"
                  >
                    <div
                      class="progress-bar"
                    >
                      <div
                        class="progress-description label-text"
                      >
                        VM3
                      </div>
                    </div>
                    <div
                      class="progress"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="100"
                        class="progress-bar"
                        role="progressbar"
                        style="width:100%"
                      />
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="0"
                        class="progress-bar progress-bar-remaining"
                        role="progressbar"
                        style="width:0%"
                      />
                    </div>
                  </div>
                  <div
                    class="progress-container progress-description-left kubevirt-consumers__bar utilization-bar-pf"
                  >
                    <div
                      class="progress-bar"
                    >
                      <div
                        class="progress-description label-text"
                      >
                        Pod1
                      </div>
                    </div>
                    <div
                      class="progress"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="80"
                        class="progress-bar"
                        role="progressbar"
                        style="width:80%"
                      />
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="20"
                        class="progress-bar progress-bar-remaining"
                        role="progressbar"
                        style="width:20%"
                      />
                    </div>
                  </div>
                  <div
                    class="progress-container progress-description-left kubevirt-consumers__bar utilization-bar-pf"
                  >
                    <div
                      class="progress-bar"
                    >
                      <div
                        class="progress-description label-text"
                      >
                        Pod3
                      </div>
                    </div>
                    <div
                      class="progress"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="70"
                        class="progress-bar"
                        role="progressbar"
                        style="width:70%"
                      />
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="30"
                        class="progress-bar progress-bar-remaining"
                        role="progressbar"
                        style="width:30%"
                      />
                    </div>
                  </div>
                  <div
                    class="progress-container progress-description-left kubevirt-consumers__bar utilization-bar-pf"
                  >
                    <div
                      class="progress-bar"
                    >
                      <div
                        class="progress-description label-text"
                      >
                        Pod2
                      </div>
                    </div>
                    <div
                      class="progress"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="40"
                        class="progress-bar"
                        role="progressbar"
                        style="width:40%"
                      />
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="60"
                        class="progress-bar progress-bar-remaining"
                        role="progressbar"
                        style="width:60%"
                      />
                    </div>
                  </div>
                  <div
                    class="progress-container progress-description-left kubevirt-consumers__bar utilization-bar-pf"
                  >
                    <div
                      class="progress-bar"
                    >
                      <div
                        class="progress-description label-text"
                      >
                        VM1
                      </div>
                    </div>
                    <div
                      class="progress"
                    >
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="20"
                        class="progress-bar"
                        role="progressbar"
                        style="width:20%"
                      />
                      <div
                        aria-valuemax="100"
                        aria-valuemin="0"
                        aria-valuenow="80"
                        class="progress-bar progress-bar-remaining"
                        role="progressbar"
                        style="width:80%"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ClusterOverview /> shallow-renders correctly 1`] = `
<Dashboard>
  <DashboardHeader>
    Cluster Dashboard
  </DashboardHeader>
  <DashboardBody
    className={null}
  >
    <Grid
      className=""
      gutter={null}
      lg={null}
      md={null}
      sm={null}
      span={null}
      xl={null}
    >
      <MediaQuery
        key="main-medium"
        maxWidth={991.9}
        values={Object {}}
      >
        <MainCards />
      </MediaQuery>
      <LeftCards />
      <MediaQuery
        key="main-large"
        minWidth={992}
        values={Object {}}
      >
        <MainCards />
      </MediaQuery>
      <RightCards />
    </Grid>
  </DashboardBody>
</Dashboard>
`;
